metrics-jersey/src/main/java/io/stargate/metrics/jersey/sgv2/TenantIdFromHostHeaderTagsProvider.java

```diff
@@ -0,0 +1,77 @@
+package io.stargate.metrics.jersey.sgv2;
+
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.jersey2.server.JerseyTagsProvider;
+import java.util.Collections;
+import javax.ws.rs.core.MultivaluedMap;
+import org.glassfish.jersey.server.ContainerRequest;
+import org.glassfish.jersey.server.monitoring.RequestEvent;
+
+/** Specialized {@link JerseyTagsProvider} used to extract Tenant Id from "Host" HTTP header. */
+public class TenantIdFromHostHeaderTagsProvider implements JerseyTagsProvider {
+  private static final String DEFAULT_TENANT_TAG_KEY = "tenant";
+
+  private final String tenantTagName;
+
+  /** Tags when tenant is unknown. */
+  private final Tags tagsForUnknown;
+
+  public TenantIdFromHostHeaderTagsProvider() {
+    this(DEFAULT_TENANT_TAG_KEY);
+  }
+
+  /**
+   * @param tenantTagName Tag name to use for Tenant Id, if any; {@code null} or empty String to
+   *     disable extraction
+   */
+  public TenantIdFromHostHeaderTagsProvider(String tenantTagName) {
+    if (tenantTagName == null || tenantTagName.isEmpty()) {
+      this.tenantTagName = null;
+      tagsForUnknown = null;
+    } else {
+      this.tenantTagName = tenantTagName;
+      tagsForUnknown = Tags.of(tenantTagName, "unknown");
+    }
+  }
+
+  @Override
+  public Iterable<Tag> httpRequestTags(RequestEvent event) {
+    return findHostTag(event);
+  }
+
+  @Override
+  public Iterable<Tag> httpLongRequestTags(RequestEvent event) {
+    return findHostTag(event);
+  }
+
+  protected Iterable<Tag> findHostTag(RequestEvent event) {
+    if (tenantTagName == null) {
+      return Collections.emptyList();
+    }
+    ContainerRequest request = event.getContainerRequest();
+    MultivaluedMap<String, String> headers = request.getHeaders();
+
+    // HTTP header names are case-insensitive but MultivaluedMap is case-sensitive
```

Comment:
Feels better for me to have extracting logic in one place and set it in the request context.. But leaving to you to decide..

Reply:
Ok, I tried to see how to reuse logic of/from `CreateStargateBridgeClientFilter` which has it but unfortunately it seems bit tricky because:

1. Extracting bit of code to get tenant id from `MultivaluedMap` would be nice but `metrics-jersey` does not depend on `sgv-service-common` nor vice versa. Worse, I don't think there's anything both depend on: `core` should not be depended on by SGv2 as it pulls in jersey-specific stuff.
2. Relying on Bridge filter would be possible by bit of refactoring (right now it only adds client in context but I could just stuff tenant id there) but would only be applied to endpoints that use annotation to get bridge client, leaving out health-check etc end points (which may or may not be a problem)
 
Now: I think that when we convert to Quarkus this is worth revisiting. But on short term I think this bit of duplication might make most sense. Not ideal but I'd prefer avoiding adding cross-dependencies as `metrics-jersey` package should go away from SGv2 once we upgrade framework.